src/jdk.jlink/share/classes/jdk/tools/jlink/internal/plugins/ZipPlugin.java

```diff
@@ -82,27 +82,22 @@ public void configure(Map<String, String> config) {
     }
 
     static byte[] compress(byte[] bytesIn, int compressionLevel) {
-        Deflater deflater = new Deflater(compressionLevel);
-        deflater.setInput(bytesIn);
-        ByteArrayOutputStream stream = new ByteArrayOutputStream(bytesIn.length);
-        byte[] buffer = new byte[1024];
-
-        deflater.finish();
-        while (!deflater.finished()) {
-            int count = deflater.deflate(buffer);
-            stream.write(buffer, 0, count);
-        }
+        try (Deflater deflater = new Deflater(compressionLevel);
+             ByteArrayOutputStream stream = new ByteArrayOutputStream(bytesIn.length)) {
 
-        try {
-            stream.close();
-        } catch (IOException ex) {
-            return bytesIn;
-        }
+            deflater.setInput(bytesIn);
 
-        byte[] bytesOut = stream.toByteArray();
-        deflater.end();
+            byte[] buffer = new byte[1024];
 
-        return bytesOut;
+            deflater.finish();
+            while (!deflater.finished()) {
+                int count = deflater.deflate(buffer);
+                stream.write(buffer, 0, count);
+            }
+            return stream.toByteArray(); // the compressed output
+        } catch (IOException e) {
+            return bytesIn; // return the original uncompressed input
```

Comment:
Can we add an `assert false : e;` assertion because this really shouldn't happen?

Reply:
I have updated the PR to add that assert and added a brief comment why we have it there. tier testing with this updated change is currently in progress.